// flow-parser/flow_parser.js
// Generated by js_of_ocaml
//# buildInfo:effects=false, kind=exe, use-js-string=true, version=5.7.2+5.7.2
(function(a){typeof
globalThis!=="object"&&(this?b():(a.defineProperty(a.prototype,"_T_",{configurable:true,get:b}),_T_));function
b(){var
b=this||self;b.globalThis=b;delete
a.prototype._T_}}(Object));(function(af){"use strict";var
yo="loc",Ni=289,Em=70416,En=69748,Kf=163,Bz=92159,v2=43587,e$="labeled_statement",Lf="&=",h5="int_of_string",Bx=110591,By=92909,q4=11559,Le="regexp",Bw=43301,q3=11703,v1=122654,h4=0xff,Nh="%ni",v0=68252,Ld=232,Bv=42785,hD="declare_variable",q2="while",El=66938,Bu=70301,Ek=124907,ok=126515,Mb=218,hC="pattern_identifier",vZ=67643,gd="export_source",vY=216,yn=64279,Lc="Out_of_memory",Bt=113788,Ng="comments",vX=126624,Ke="win32",gc="object_key_bigint_literal",Kd=185,Nf=283,q1=123214,ja="constructor",ym=69955,gb="import_declaration",Ei=68437,Ej="Failure",p4="Unix.Unix_error",vW=64255,vV=42539,yl=110579,e_="export_default_declaration",dY="jsx_attribute_name",q0=11727,vU=43002,oj=126500,hB="component_param_pattern",Kc="collect_comments_opt",e9="match_unary_pattern",dX="keyof_type",Ma="Invalid binary/octal ",Lb="range",pd=170,h8="false",vT=43798,Kb=", characters ",hA="object_type_property_getter",Bs=65547,L$=299,vS=126467,Br=65007,Bq=42237,yk=8318,Bp=71215,dW="object_property_type",e8="type_alias",Eh=67742,hz="function_body",La=304,Bo=68111,pc=120745,Bn=71959,p3=43880,K$="Match_failure",vR=280,dV="type_cast",dU=109,hZ="void",Eg="generator",Bl=125124,Bm=101589,p2=94179,L_=">>>",pb=70404,e7="optional_indexed_access_type",a_="argument",ga="object_property",hy="object_type_property",Ef=67004,vQ=42783,Bk=68850,L9="@",yi=43741,yj=43487,pa="object",L8="end",oi=126571,Bj=71956,L7=208,Ed=126566,Ee=67702,Ne="EEXIST",e6="this_expression",L6=203,Bi=11507,vP=113807,oh=119893,yh=42735,lR="rest",e5="null_literal",lZ="protected",Bh=43615,aZ=8231,yg=68149,vO=73727,Ec=72348,vN=92995,jj=224,vM=11686,yf=43013,e4="assignment_pattern",Bg=12329,f$="function_type",ku=192,e3="jsx_element_name",Eb=70018,hx="catch_clause_pattern",p1=126540,f_="template_literal",vL=120654,D$=68497,Ea=67679,hw="readonly_type",vJ=68735,vK="<",o$=": No such file or directory",D_=66915,Nd="!",dT="object_type",ye=43712,qZ=64297,yd=183969,Bf=43503,D9=67591,jz=65278,yc=67669,hv="for_of_assignment_pattern",ly="`",D8=11502,hu="catch_body",vI=42191,iC=-744106340,og=182,jy=":",Ka="a string",Be=65663,D7=66978,D6=71947,qY=43519,yb=71086,vH=125258,D5=12538,e2="expression_or_spread",K_="Printexc.handle_uncaught_exception",p0=69956,of=120122,o_=247,J$=231,vG=" : flags Open_rdonly and Open_wronly are not compatible",ht="statement_fork_point",K9=710,J_=-692038429,bM="static",ya=55203,vE=64324,vF=64111,J9="!==",D4=120132,vD=124903,lx="class",K8=222,f9="pattern_number_literal",hY="kind",vC=71903,dS="variable_declarator",dR="typeof_expression",x$=126627,Bd=70084,Nc=228,qX=70480,e1="class_private_field",Bc=239,oe=120713,cz=0xffff,e0="private_name",Bb=43137,f8="remote_identifier",vB=70161,eZ="label_identifier",Ba="src/parser/statement_parser.ml",vA=8335,A$=19903,D3=64310,jo="_",dQ="for_init_declaration",K7="infer",D1=64466,D2=43018,J8="tokens",A_=92735,D0=66954,vz=65473,DZ=70285,f7="sequence",DY="compare: functional value",vy=69890,lY=1000,vx=65487,DX=42653,J7="\\\\",L5="%=",hs="match_member_pattern_base",A9=72367,dP="function_rest_param",J6="/static/",vw=124911,A8=65276,pZ=126558,vv=11498,J5=137,f6="export_default_declaration_decl",vu="cases",od=126602,dO="jsx_child",bL="continue",vt=42962,Nb="importKind",ai=122,kt="Literal",hr="pattern_object_property_identifier_key",x_=42508,iF="in",A7=55238,x9=67071,A6=70831,A4=72161,A5=67462,J4="<<=",vs=43009,vr=66383,oc=67827,A3=72202,x8=69839,DW=66775,J3="-=",ji=8202,DV=70105,A2=120538,dN="for_in_left_declaration",A1="rendersType",pY=126563,DU=70708,ob=126523,J2=166,hq="match_",J1=202,A0=110951,h3="component",o9=126552,DT=66977,Na=213,hp="enum_member_identifier",L4=210,ho="enum_bigint_body",K6=">=",AY=126495,AZ="specifiers",K5=-88,AX="=",x7=65338,lw="members",K4=309,vp=123535,vq=43702,x6=72767,jx="get",AW=126633,oa=126536,AU=94098,AV="types",K3=273,DS=113663,K2="Internal Error: Found private field in object props",eY="jsx_element",AT=70366,x5=110959,o8=120655,M_=305,M$="trailingComments",iA=24029,DR=-100,bo="yield",dM="binding_pattern",hn="typeof_identifier",J0="ENOTEMPTY",x4=-104,L3=295,o7=126468,AS=1255,vo=120628,eX="pattern_object_property_string_literal_key",DQ=8521,JY="leadingComments",JZ=8204,iB="@ ",x3=70319,hX="left",L2=188,pX="case",AR=19967,pW=42622,x2=43492,DP=113770,AP=42774,AQ=183,o6=8468,f5="class_implements",n$=126579,ka="string",K1=211,aR=-48,DO=69926,DN=123213,dL="if_consequent_statement",AO=124927,ks="number",vn=126546,x1=68119,AN=70726,o5=70750,DM=65489,M9="SpreadElement",L1="callee",JX=193,AM=70492,AL=71934,JW=164,AK=110580,DL=12320,pV="any",bv="/",hm="type_guard",ax="body",L0=178,bE="pattern",K0="comment_bounds",f4="binding_type_identifier",vm=187,eW="pattern_array_rest_element_pattern",n_="@])",DJ=12543,DK=11623,M8="start",vl=67871,bu="interface",AJ=8449,vk=67637,vj=42961,qW=120085,DI=126463,JV="alternate",LZ=-1053382366,x0=70143,JU="--",AI=68031,hl="jsx_expression",eV="type_identifier_reference",pU=11647,DH="proto",cg="identifier",AH=43696,cf="raw",DG=126529,AG=11564,qV=126557,DF=64911,o4=67592,xZ=43493,DE=215,DD=110588,lI=461894857,vi=92927,AF=67861,DB=119980,DC=43042,AE=66965,xY=67391,kk="computed",LY="unreachable jsxtext",Dz=71167,DA=42559,vh=72966,JT=180,xX=197,n9=64319,JR=169,JS="*",jw=129,vg=66335,lv="meta",AD=43388,o3=94178,b1="optional",pT="unknown",Dy=120121,Dx=123180,o2=8469,AC=68220,LX="|",Dw=43187,AB=94207,vf=124895,qU=120513,ve=42527,jh=8286,Dv=94177,lH="var",f3="component_type_param",Du=66421,xW=92991,AA=68415,dK="comment",hk="match_pattern_array_element",jg=244,pS="^",xV=173791,KZ=136,Az=42890,vd="ENOTDIR",Dt="??",Ax=43711,Ay=66303,Ds=113800,vc=42239,Dr=12703,hi="variance_opt",hj="+",JQ=">>>=",n8="mixed",xU=65613,Aw=73029,xT=68191,KY="*=",pR=8487,vb=8477,f2="toplevel_statement_list",n6="never",n7="do",iH=125,va=72249,JP="Pervasives.do_at_exit",JO="visit_trailing_comment",dJ="jsx_closing_element",eU="jsx_namespaced_name",u$=124908,Av=126651,f1="component_declaration",Au=0xf,f0="interface_type",eT="function_type_return_annotation",xS=64109,o1=65595,qT=126560,u_=110927,pP=65598,pQ=8488,h7="`.",M7=175,pO="package",qS="else",o0=120771,u9=68023,KX="fd ",jn=8238,n4=888960333,n5=119965,u8=42655,hh="match_object_pattern",u7=11710,Dq=119993,hg="boolean_literal",dI="statement_list",fZ="function_param",dH="match_as_pattern",eS="pattern_object_property_bigint_literal_key",pN=69959,u6=120485,JN=0xf0,u5=191456,fY="declare_enum",pM=120597,pL=70281,fW="type_annotation",fX="spread_element",n3=126544,xR=120069,h2="key",xQ=43583,u3="out",u4="\n",KW="**=",eR="pattern_object_property_pattern",xP="e",u2=72712,M6="Internal Error: Found object private prop",At="ENOENT",u1=-42,eQ="jsx_opening_attribute",As=67646,eP="component_type",xO=64296,xN=43887,LW="Division_by_zero",LV="EnumDefaultedMember",hf="typeof_member_identifier",u0=43792,dG="match_member_pattern_property",fV="declare_export_declaration_decl",Dp=93026,fU="type_annotation_hint",uZ=42887,Ar=43881,Aq=43761,pK=8526,j$=119,Do=43866,xL=72847,xM=8348,bt=101,Ap=94026,pJ=72272,JM="src/parser/flow_lexer.ml",Ao=120744,jf=8191,kj="implies",pI=255,qR=11711,he="match_unary_pattern_argument",uY=71235,LU=288,pH=68116,JL=261,an=100,fT="match_expression",hd="enum_body",pG=1114111,hc="assignment",uX=71955,n2=43260,eO="pattern_array_e",Dn=126583,JK="prefix",fS="class_body",lQ="shorthand",qQ=171,uW=66256,oZ=-97,LT=" =",Dl=94032,Dm=42606,fR="match_case",xK=71839,n1=120134,xJ=55291,An=92862,xI=43019,Dk=126543,kd="function",Am=111355,Dj=11389,uV=70753,uU=43249,Di=64829,n0="line",fQ="function_declaration",qP="undefined",M5="([^/]+)",xH=110947,Dh=70002,KV="Cygwin",dF="as_expression",Dg=12591,qO=64285,Df=2048,uT=73112,oY=126589,M4=225,oX=43259,xG=72817,oW=64318,M3=172,LS=209,ha="match_binding_pattern",hb=" ",eN="import_source",lG="delete",LR="Enum `",qN=126553,Al=67001,jm="default",xE=11630,xF=206,eM="enum_bigint_member",Ak=67504,qM=67593,xD=113791,M2="MatchObjectPatternProperty",uS=69572,g$="typeof_type",M0=212,M1="%i",dE="function_this_param",xC=72329,iE="0x",jv=8239,De=75075,MZ=277,LQ=0xdfff,eL="pattern_bigint_literal",Dd=12341,MY=201,i$="hook",JJ=": closedir failed",Dc=42959,nZ=119970,xB=43560,LP="||=",g_="member_private_name",uR=120570,eK="object_key_identifier",nY=223,KU="Not_found",MX=230,eJ="jsx_element_name_member_expression",g9="string_literal",Aj=120596,Db=43807,Ai=69687,uQ=63743,qL=72192,dD="member_property",xA=43262,g8="class_declaration",JI="renders*",KT="%Li",xz=126578,fP="jsx_attribute",kr=254,bF="empty",lX="label",g7="object_internal_slot_property_type",oV=120133,uP=43359,bK="predicate",LO="??=",C$=43697,Da=-43,fO="default_opt",MW="the start of a statement",KS=312,uO=67826,fM="object_",fN="class_element",qK=11631,oU=70855,eI="opaque_type",fL="number_literal",MV=", ",oT=8319,qJ=120004,oS=133,fK="type_params",dC="pattern_object_rest_property",bk="import",Ah=72000,C9=67413,C_=12343,Ag=70080,g6="intersection_type",aY=-36,uN=70005,qI="properties",xx=11679,xy=8483,Af=110587,KR=43520,eH="computed_key",uM=207,g5="class_identifier",uL="Invalid number ",dB="function_param_pattern",je=12288,Ae=113817,xw=70730,Ad=178207,pF=71236,MU=167,dA="object_indexer_property_type",uK=64286,LM="TypeAnnotation",LN=220,fJ="type_identifier",dy="spread_property",dz="jsx_attribute_value_expression",Ac=126519,qH=70108,oR=126,nX=42999,iG="prototype",C8=" : flags Open_text and Open_binary are not compatible",JH="**",oQ=43823,C7=": Not a directory",eG="render_type",qG=72349,ki="test",C6=43776,uI=92879,uJ=11263,KQ=241,uH=93052,eF="nullable_type",Ab=43704,C5=64321,KP="Property",Aa=72191,MT=165,lP="instanceof",C4=69247,KO=302,bU="name",oP=126634,z$=8516,pE="typeArguments",z_=71127,g4="jsx_spread_attribute",xv=66559,C3=44031,z9=43645,aO=8233,xt=71494,xu="opaque",nW=72967,xs=70106,eE="logical",JG="@[%s =@ ",lO="0o",oO=126554,uG=71351,qF=8484,xr=72242,nV=120687,kc=252,xq=183983,lW="%S",dx="function_this_param_type",qE="decorators",uF=43255,dw="catch_clause",bP="-",z8=67711,MS=": file descriptor already closed",oN=64311,qD=120539,uE="arguments",oM=73062,z7=173823,uD=42124,C2=72095,uB=125259,uC=42969,nU=70280,LL=12520,uA=69749,xp=70066,fI="binary",fH="for_in_statement",z6=43010,JF="^=",xo=126570,eD="for_statement",oL=126584,eC="function_return_annotation",C1=72144,z5=8505,eB="class_expression",uz=120076,z4=69807,z3=40981,CZ=-24976191,C0=72768,xn=126550,qC='"',fG="call_type_arg",LK="f",i_="this",nT=126628,JE="===",KN=0xdc00,g3="declare_module_exports",CY=120512,b2=105,ux=119974,uy=71450,uw=71942,LJ=195,pD=120629,JD="/=",KM=">>",eA="declare_interface",LI=4096,dv="pattern_array_rest_element",xm=71338,oK=126520,fF="as_const_expression",MR="Popping lex mode from empty stack",MQ="renders?",xl=68405,ez="member",g2="class_extends",jd=12287,pC=126590,uv=66377,iz="async",g1="pattern_array_element",kh=240,z2=69864,jl="readonly",z1=70460,ut=120779,uu=66378,du="new_",oJ=126551,g0="pattern_object_rest_property_pattern",ey="for_statement_init",us=43595,MP=293,pB=68296,xj=120712,xk=64217,z0=69295,JC="||",zZ=";",ur=70461,xi=66939,MO="collect_comments",CX=279,dt="generic_type",zY=68295,zX=44002,pA=72162,fE="object_call_property_type",qB=8305,oI=119995,pz="with",gZ="class_property",KL="qualification",ew="jsx_attribute_name_namespaced",ex="if_statement",ds="typeof_qualified_identifier",MN=238,xh=65615,KK=176,aP="expression",qA=126559,fD="jsx_attribute_value",gX="<2>",gY="component_param",qz="Map.bal",py=132,CW=70412,xg=70440,KJ="<<",px="finally",KI="v",fC="syntax_opt",dr="meta_property",zW=12447,xf=67514,JB=260,pw=12448,dq="object_mapped_type_property",jc="operator",KH="closedir",fB="unary_expression",CU=126588,CV=70851,fA="export_batch_specifier",kq="renders",KG=226,zV=73111,xe=221,e="",xb=66927,xc=64967,xd="elements",xa=67640,zU=43754,gW="declare_export_declaration",w_=-26065557,w$=65855,lF="boolean",h6="typeof",uq=124902,LH=139,CT=65629,MM=0xe0,CS=43123,nS=70449,zT=12735,aG=107,qy=11719,ML="!=",dp="call_type_args",kp="asserts",iy=-46,zS="namespace",dm="match_pattern",dn="for_of_statement_lhs",oH=126504,CR=69505,pv="for",up=72703,pt=120127,pu=43471,CQ=93047,KF="Undefined_recursive_module",KE=2147483647,gV="template_literal_element",JA="Unexpected ",zQ=101631,zR=65497,ps=68120,fz="import_default_specifier",cG="array",LG="expressions",CP=110930,KD=204,gT="while_",gU="function_rest_param_type",iD=0x3f,CO=77808,MK="Unexpected token `",V=114,fy="pattern_object_p",w9=65140,uo=123190,dl="pattern_object_property_number_literal_key",lE="enum",fx="conditional_type",bn=113,dk="array_type",KC="minus",un=43790,dj="do_while",w8=11567,CN=11694,lD=256,CM=119976,gS="component_body",bj=111,zP=177976,LF=-56,nR=67644,w7=73439,lN=951901561,LD="?",LE=")",nQ=43867,oG=65575,CL=69445,MJ="FunctionTypeParam",qx=119996,um=65019,di="conditional",zO=11505,Jz=135,w6=71295,zN=12799,CK=67382,ev="type_guard_annotation",dh="object_key_computed",cy=123,gR="pattern_object_property_key",zM=119892,w5=67505,zL=66962,dg="with_",ul=43273,df="interface_declaration",nP="bool",zK=71945,CJ="declaration",CI=11519,lV=">",zJ=66771,oF="}",KB=8472,uk=43014,fw="declare_function",aa=127,CH="RestElement",zI=190,uj=8467,Kz="module",oE=126522,KA="Sys_blocked_io",gQ="jsx_opening_element",fv="object_key_number_literal",MI="|=",Ky="mixins",w4=205,LC=217,qw="if",MH="+=",de="match_object_pattern_property_key",et="match_rest_pattern",eu="export_named_declaration_specifier",pq="try",pr="_bigarr02",CF=70479,cF="right",CG=245,zH=11718,gP="tuple_labeled_element",LB="TypeParameterInstantiation",CE="mkdir",CD=71999,zG=870530776,LA="@[",Kx=-908856609,Lz=331416730,zF=11670,CC=66735,ui=43709,qv=43642,uh=67002,zE=69375,fu="function_body_any",CB=119807,MG="Assert_failure",es="function_identifier",CA=65479,lu=131,ju="new",gO="for_of_left_declaration",ug=120084,w3=100343,Cz=73030,pp=70452,Kw=134,uf=253,Cy=42954,MF=227,gN="jsx_member_expression_object",er="class_property_value",w2=120144,Cx=66994,j_="set",Cw=126498,gM="tuple_element",dd="arg_list",zD=65481,w1=8511,ue=42964,Cv=11492,kb=-25,qu=126555,ud=71039,Cu="exportKind",dc="program",zC=70187,Ly=173,ce="as",kg=124,Lx="visit_leading_comment",uc=110575,eq="class_",Ct=72440,Cs=67897,ME=235,w0=8543,Kv=141,eo=120,ep="match_object_pattern_property",lt=1024,wZ=101640,Ku=1027,MD=236,j9=246,Kt="(",ub=66511,db="regexp_literal",zB=65574,zA=43513,ua=43695,Jy="&&",nO=11558,wY=66503,Cr=93071,da="pattern_expression",Cq=65381,qt=126538,Cp=12292,gL="import_namespace_specifier",Co=67583,t$=120137,wX=69622,Cn=120770,Cm=71131,jb=8287,zy=110590,zz=65135,wW="Fatal error: exception ",lU=118,Cl=181,qs=11687,a0="camlinternalFormat.ml",Cj=72959,Ck=249,en="union_type",Ks=8206,zx=73064,Ci=70271,wV=92728,nN=65344,oD=11695,gK="class_decorator",MC="the end of an expression statement (`;`)",t_=177983,zw=8457,Lw=931,Ch=66499,t9=94175,Jx="#",zv="Identifier",gJ="for_in_statement_lhs",c$="pattern_string_literal",po=70302,qr=126496,zu=66461,Cg=82943,nM=8450,t8=72271,Ce=70853,Cf="of",MB="Stack_overflow",lr="hasUnknownMembers",ls="a",MA=153,em="variable_declarator_pattern",zt=73061,t7=77711,oC=64317,zs=73097,gI="enum_declaration",zr=66966,qq=189,wU=119964,ft="type_param",cv=782176664,oB=65535,Lv=-10,zq=64433,oA=43815,nK=94031,nL=73065,wT=69958,qp="property",c_="jsx_children",el="member_property_identifier",t6=42537,ix="const",zp=70278,ek="enum_string_member",lM="local",fs="jsx_element_name_identifier",t5=68223,oz="\x01\x02",wS=119967,pn=119994,t4=66993,ej="jsx_member_expression_identifier",pm="explicitType",wR=67589,Cd=65597,t3="exported",zo=94111,zn=113775,ei="object_spread_property_type",wQ=64847,gH="component_identifier",fr="class_implements_interface",Mz=162,Jw=243,t2=12783,Jv="Fatal error: exception %s\n",oy=120093,lq="column",eh="component_rest_param",tZ=70451,t0=70312,t1=69967,qo=70279,wP=66463,tY=92975,nJ=70286,c8="pattern_object_property_computed_key",c9="object_key_string_literal",tX="jsError",fq="type_args",tW=8304,Lu="==",jt=115,gG="declare_component",zm=120092,Cc=43638,tV=66811,wO=43334,wN=66863,tU=77823,Lt=143,gF="optional_call",zl=126562,pl=70162,bT=104,zk=66963,js="await",nI=70107,bm="0",wL=72250,wM=8507,zj=100351,qn="AssignmentPattern",eg="type",Ls="%u",ef="function_expression_or_method",zi=43470,My=242,Ju="camlinternalMod.ml",ee="match_or_pattern",zh=72750,Cb=69414,tT=65370,c7="syntax",Lr=0x7ff0,Ca=42963,Mx="End_of_file",zg=12294,B$=8471,Jt="elementType",B_=43782,Mw="++",wK=43641,B9=71944,wJ=126601,zf=78894,B8=-45,jr="null",ze=177,Kr="satisfies",B7=131071,c6="import_specifier",gE="class_method",ed="type_",B6=126514,zd=8454,Kq="inexact",tS=67807,tR=8525,tQ=65470,B5=71352,fp="tuple_spread_element",Js=219,zc="abstract",wI=73458,bO="return",lT=65536,nH=126548,ec="array_element",zb=-253313196,nG=186,ox="catch",eb="infer_type",B4=12295,Mv="Invalid legacy octal ",tO=69762,tP=43311,za=65437,c5="variable_declaration",Jr=-696510241,gD="function_params",y$=64316,Mu=311,ow=11565,Mt="infinity",wH="@]",B3=65908,gC="extends",tM=66204,tN=43784,y_=11742,qm=126503,bN="debugger",y9=70457,h1=-86,lL=912068366,B2=68786,qk="keyof",ql=69415,wG=12686,cD=127343600,fo="declare_type_alias",Jq="the",Kp=233,ea="jsx_element_name_namespaced",wF=72283,Ms=161,d$="function_param_type",ch=0x80,y8=-673950933,pk=126591,Mr="Sys_error",y7=74649,tL=74862,lC="is",tK=43738,B1=68479,Lq=196,qj=70854,gB="enum_boolean_member",pj=72163,y6=92783,Jp=281,gA="component_param_name",wE=68863,cx=0x8000,Mq=0x800,wD=64284,Ko="@{",y5="\x01\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\x01",pi=8455,gz="update_expression",y4=65500,lS="from",wC=68447,nF=12592,wB=92766,Lp=">>=",a9=110,B0=66431,tJ=43586,d_="jsx_identifier",wA=" : file already exists",aA=128,BZ=71958,tI=66717,c4="enum_boolean_body",y3=64262,b="id",c3="component_renders_annotation",BY=42888,tH=8584,tG=73008,c2="enum_symbol_body",d9="declare_namespace",ov=72713,y2=55215,c1="object_property_value_type",fn="for_in_assignment_pattern",nE=8485,wz=43395,Mp=229,hW="true",y1=43743,fm="enum_number_member",Jo=234,y0=72969,Lo="expected *",cC=102,Mo=200,lB="symbol",jq="source",gy="tparam_const_modifier",wy=43714,gx="jsx_fragment",fl="jsx_attribute_name_identifier",lp="public",wx=43442,d8="pattern_object_property",BX=65786,tE=70783,tF=43713,ww=72160,Ln="*-/",cZ="export_named_specifier",c0="arrow_function",wv=122623,qi=70006,Kn="${",yZ=43814,fk="generic_qualified_identifier_type",Mn=199,d7="jsx_spread_child",qg=8489,qh=184,Jn=2047,wu=66955,d6="try_catch",yY=70497,Mm=237,wt=67431,yX=125183,Jm=-602162310,cB="params",ws="consequent",yW=68029,yV=67829,BW=68095,gw="enum_string_body",tC=93823,tD=68351,wr=65495,fj="declare_module",d5="match_as_pattern_target",d4="body_expression",yU=66175,qf=191,ou=70441,qd=65141,qe="&",gv="super_expression",ot=126564,tB=72105,yT="fs",bS="throw",tA=68287,wq=67839,i9=116,wp=110882,BV=69404,tz=123197,i8=65279,j8="src/parser/type_parser.ml",BU=68115,Lm=159,qc=126547,nD=126556,yS=73055,gu="member_property_expression",fi="enum_defaulted_member",yR=43071,BT=11726,gt="component_type_rest_param",wo=68607,gs="object_key",Ll=160,bl="variance",BR=70655,BS=70414,kf="super",ty=123583,yQ=65594,lK="method",wn=73648,lA=121,BQ=93951,fh="pattern_array_element_pattern",wl=43764,wm=42993,os=120145,yP=74879,Ml=168,or=8486,yO=72001,ff="tagged_template",fg="module_ref_literal",BP=65312,jk="implements",tx=43700,wk=120003,Mk="Invalid_argument",gr=0xffffff,tw=83526,nC=69744,oq=12336,gq="switch_case",Km=-61,cY="optional_member",tv=64274,qb=64322,nB=126530,wj=71998,ph=72970,BO=13311,wi=73647,yN=120074,ke="let",Mj="global",d2="expression_statement",d3="component_type_params",wh=512,yL=69634,yM=67461,BN=123627,tu=64913,Jl="children",Kk="PropertyDefinition",Kl=1026,Mi="%li",gp="declare_class",tt=43258,d1="indexed_access_type",yJ=124926,h0=112,yK="b",gn="predicate_expression",go="if_alternate_statement",lz="private",Lk=-594953737,ts=140,yI="nan",wf=72103,op=11735,d0="statement",wg="rmdir",nA=66512,BM="match",Mh=198,yH=11734,gm="import_named_specifier",tr=69599,yG=68799,BL=194559,cX="match_array_pattern",Mg=174,gk="function_",gl="bigint_literal",ah=248,nz=67638,qa=126539,yF=11557,Jk=214,we=5760,bR="break",cE="block",dZ="match_member_pattern",yE=123565,wd=66815,ap="value",Jj=1039100673,yD=69746,BK=70448,wc=74751,gj="init",BJ=69551,p$=65548,gi="jsx_member_expression",oo=68096,hV=108,pg=126521,BI=71487,gh="match_statement",tq=178205,tp=12548,Lj=" : is a directory",cw=".",yC=12348,ko=-835925911,bi="typeParameters",BH=66855,aQ="typeAnnotation",jp="bigint",fe="jsx_attribute_value_literal",BG=194,Mf="T_JSX_TEXT",to=68466,on=126537,Ji=67714067,tn=69487,ny="export",BF=43822,p_=126499,tm=55242,cW="member_type_identifier",Kj=138,yB=71679,lJ=130,BE=12438,wb=119969,p9=12539,yA=119972,Me=",",v$=71423,wa="index out of bounds",cA=106,j7="%d",Ki="T_RENDERS_QUESTION",p8=120571,om="returnType",yz=69423,pf=120070,Li="%",kn=117,Md=179,tl="EBADF",yy=93759,p7=64325,gg="component_params",yx=66517,tk=67423,tj=605857695,v_=43518,Lh=251,cV="for_of_statement",v9=71983,Kh="~",yw=12442,bQ="switch",BD=66207,p6=126535,Jh="&&=",v8=69289,yv=71723,gf="generic_identifier_type",yu=126619,fd="object_type_property_setter",BC=70418,Mc="<=",v7=125251,BB=11702,fc="enum_number_body",km=250,th=124910,ti=69297,yt=67455,ys=42511,cU="ts_satisfies",Lg=268,v6=68324,p5="an identifier",tg=126534,be=103,v5=120126,Kg=274,kl=449540197,lo="declare",v4=68899,tf=126502,fb="function_expression",Jg=142,BA=123135,te=67967,yr=120487,td=120686,cT="export_named_declaration",yq=66348,pe=119981,yp=12352,ge="tuple_type",v3=68680,ol="target",fa="call";function
awU(a,b,c,d,e){if(d<=b)for(var
f=1;f<=e;f++)c[d+f]=a[b+f];else
for(var
f=e;f>=1;f--)c[d+f]=a[b+f];return 0}function
awV(a){var
c=[0];while(a!==0){var
d=a[1];for(var
b=1;b<d.length;b++)c.push(d[b]);a=a[2]}return c}function
awW(a,b,c){var
d=new
Array(c+1);d[0]=0;for(var
e=1,f=b+1;e<=c;e++,f++)d[e]=a[f];return d}function
q6(a,b,c){if(a[1]===b){a[1]=c;return 1}return 0}function
awX(a,b){var
c=a[1];a[1]+=b;return c}function
kw(a){return a[1]}function
NT(a){var
b=af.process;if(b&&b.env&&b.env[a]!=undefined)return b.env[a];if(af.jsoo_static_env&&af.jsoo_static_env[a])return af.jsoo_static_env[a]}var
EJ=0;(function(){var
c=NT("OCAMLRUNPARAM");if(c!==undefined){var
b=c.split(Me);for(var
a=0;a<b.length;a++)if(b[a]==yK){EJ=1;break}else if(b[a].startsWith("b="))EJ=+b[a].slice(2);else
continue}}());var
bq=[0];function
aw9(a,b){if(!a.js_error||b||a[0]==ah)a.js_error=new
af.Error("Js exception containing backtrace");return a}function
x(a,b){return EJ?aw9(a,b):a}function
axH(a,b){throw x([0,a,b])}function
b5(a){return a}function
EI(a,b){axH(a,b5(b))}function
aT(a){EI(bq.Invalid_argument,a)}function
Nn(a){switch(a){case
7:case
10:case
11:return 2;default:return 1}}function
Nl(a,b){var
c;switch(a){case
0:c=Float32Array;break;case
1:c=Float64Array;break;case
2:c=Int8Array;break;case
3:c=Uint8Array;break;case
4:c=Int16Array;break;case
5:c=Uint16Array;break;case
6:c=Int32Array;break;case
7:c=Int32Array;break;case
8:c=Int32Array;break;case
9:c=Int32Array;break;case
10:c=Float32Array;break;case
11:c=Float64Array;break;case
12:c=Uint8Array;break}if(!c)aT("Bigarray.create: unsupported kind");var
d=new
c(b*Nn(a));return d}function
q7(a){var
d=a.length,c=1;for(var
b=0;b<d;b++){if(a[b]<0)aT("Bigarray.create: negative dimension");c=c*a[b]}return c}var
ND=Math.pow(2,-24);function
NN(a){throw a}function
NP(){NN(bq.Division_by_zero)}function
N(a,b,c){this.lo=a&gr;this.mi=b&gr;this.hi=c&cz}N.prototype.caml_custom="_j";N.prototype.copy=function(){return new
N(this.lo,this.mi,this.hi)};N.prototype.ucompare=function(a){if(this.hi>a.hi)return 1;if(this.hi<a.hi)return-1;if(this.mi>a.mi)return 1;if(this.mi<a.mi)return-1;if(this.lo>a.lo)return 1;if(this.lo<a.lo)return-1;return 0};N.prototype.compare=function(a){var
b=this.hi<<16,c=a.hi<<16;if(b>c)return 1;if(b<c)return-1;if(this.mi>a.mi)return 1;if(this.mi<a.mi)return-1;if(this.lo>a.lo)return 1;if(this.lo<a.lo)return-1;return 0};N.prototype.neg=function(){var
a=-this.lo,b=-this.mi+(a>>24),c=-this.hi+(b>>24);return new
N(a,b,c)};N.prototype.add=function(a){var
b=this.lo+a.lo,c=this.mi+a.mi+(b>>24),d=this.hi+a.hi+(c>>24);return new
N(b,c,d)};N.prototype.sub=function(a){var
b=this.lo-a.lo,c=this.mi-a.mi+(b>>24),d=this.hi-a.hi+(c>>24);return new
N(b,c,d)};N.prototype.mul=function(a){var
b=this.lo*a.lo,c=(b*ND|0)+this.mi*a.lo+this.lo*a.mi,d=(c*ND|0)+this.hi*a.lo+this.mi*a.mi+this.lo*a.hi;return new
N(b,c,d)};N.prototype.isZero=function(){return(this.lo|this.mi|this.hi)==0};N.prototype.isNeg=function(){return this.hi<<16<0};N.prototype.and=function(a){return new
N(this.lo&a.lo,this.mi&a.mi,this.hi&a.hi)};N.prototype.or=function(a){return new
N(this.lo|a.lo,this.mi|a.mi,this.hi|a.hi)};N.prototype.xor=function(a){return new
N(this.lo^a.lo,this.mi^a.mi,this.hi^a.hi)};N.prototype.shift_left=function(a){a=a&63;if(a==0)return this;if(a<24)return new
N(this.lo<<a,this.mi<<a|this.lo>>24-a,this.hi<<a|this.mi>>24-a);if(a<48)return new
N(0,this.lo<<a-24,this.mi<<a-24|this.lo>>48-a);return new
N(0,0,this.lo<<a-48)};N.prototype.shift_right_unsigned=function(a){a=a&63;if(a==0)return this;if(a<24)return new
N(this.lo>>a|this.mi<<24-a,this.mi>>a|this.hi<<24-a,this.hi>>a);if(a<48)return new
N(this.mi>>a-24|this.hi<<48-a,this.hi>>a-24,0);return new
N(this.hi>>a-48,0,0)};N.prototype.shift_right=function(a){a=a&63;if(a==0)return this;var
c=this.hi<<16>>16;if(a<24)return new
N(this.lo>>a|this.mi<<24-a,this.mi>>a|c<<24-a,this.hi<<16>>a>>>16);var
b=this.hi<<16>>31;if(a<48)return new
N(this.mi>>a-24|this.hi<<48-a,this.hi<<16>>a-24>>16,b&cz);return new
N(this.hi<<16>>a-32,b,b)};N.prototype.lsl1=function(){this.hi=this.hi<<1|this.mi>>23;this.mi=(this.mi<<1|this.lo>>23)&gr;this.lo=this.lo<<1&gr};N.prototype.lsr1=function(){this.lo=(this.lo>>>1|this.mi<<23)&gr;this.mi=(this.mi>>>1|this.hi<<23)&gr;this.hi=this.hi>>>1};N.prototype.udivmod=function(a){var
d=0,c=this.copy(),b=a.copy(),e=new
N(0,0,0);while(c.ucompare(b)>0){d++;b.lsl1()}while(d>=0){d--;e.lsl1();if(c.ucompare(b)>=0){e.lo++;c=c.sub(b)}b.lsr1()}return{quotient:e,modulus:c}};N.prototype.div=function(a){var
b=this;if(a.isZero())NP();var
d=b.hi^a.hi;if(b.hi&cx)b=b.neg();if(a.hi&cx)a=a.neg();var
c=b.udivmod(a).quotient;if(d&cx)c=c.neg();return c};N.prototype.mod=function(a){var
b=this;if(a.isZero())NP();var
d=b.hi;if(b.hi&cx)b=b.neg();if(a.hi&cx)a=a.neg();var
c=b.udivmod(a).modulus;if(d&cx)c=c.neg();return c};N.prototype.toInt=function(){return this.lo|this.mi<<24};N.prototype.toFloat=function(){return(this.hi<<16)*Math.pow(2,32)+this.mi*Math.pow(2,24)+this.lo};N.prototype.toArray=function(){return[this.hi>>8,this.hi&h4,this.mi>>16,this.mi>>8&h4,this.mi&h4,this.lo>>16,this.lo>>8&h4,this.lo&h4]};N.prototype.lo32=function(){return this.lo|(this.mi&h4)<<24};N.prototype.hi32=function(){return this.mi>>>8&cz|this.hi<<16};function
axe(a,b){return new
N(a&gr,a>>>24&h4|(b&cz)<<8,b>>>16&cz)}function
Ey(a){return a.hi32()}function
Ez(a){return a.lo32()}function
l1(){aT(wa)}var
awZ=pr;function
iI(a,b,c,d){this.kind=a;this.layout=b;this.dims=c;this.data=d}iI.prototype.caml_custom=awZ;iI.prototype.offset=function(a){var
c=0;if(typeof
a==="number")a=[a];if(!(a
instanceof
Array))aT("bigarray.js: invalid offset");if(this.dims.length!=a.length)aT("Bigarray.get/set: bad number of dimensions");if(this.layout==0)for(var
b=0;b<this.dims.length;b++){if(a[b]<0||a[b]>=this.dims[b])l1();c=c*this.dims[b]+a[b]}else
for(var
b=this.dims.length-1;b>=0;b--){if(a[b]<1||a[b]>this.dims[b])l1();c=c*this.dims[b]+(a[b]-1)}return c};iI.prototype.get=function(a){switch(this.kind){case
7:var
d=this.data[a*2+0],b=this.data[a*2+1];return axe(d,b);case
10:case
11:var
e=this.data[a*2+0],c=this.data[a*2+1];return[kr,e,c];default:return this.data[a]}};iI.prototype.set=function(a,b){switch(this.kind){case
7:this.data[a*2+0]=Ez(b);this.data[a*2+1]=Ey(b);break;case
10:case
11:this.data[a*2+0]=b[1];this.data[a*2+1]=b[2];break;default:this.data[a]=b;break}return 0};iI.prototype.fill=function(a){switch(this.kind){case
7:var
c=Ez(a),e=Ey(a);if(c==e)this.data.fill(c);else
for(var
b=0;b<this.data.length;b++)this.data[b]=b%2==0?c:e;break;case
10:case
11:var
d=a[1],f=a[2];if(d==f)this.data.fill(d);else
for(var
b=0;b<this.data.length;b++)this.data[b]=b%2==0?d:f;break;default:this.data.fill(a);break}};iI.prototype.compare=function(a,b){if(this.layout!=a.layout||this.kind!=a.kind){var
f=this.kind|this.layout<<8,g=a.kind|a.layout<<8;return g-f}if(this.dims.length!=a.dims.length)return a.dims.length-this.dims.length;for(var
c=0;c<this.dims.length;c++)if(this.dims[c]!=a.dims[c])return this.dims[c]<a.dims[c]?-1:1;switch(this.kind){case
0:case
1:case
10:case
11:var
d,e;for(var
c=0;c<this.data.length;c++){d=this.data[c];e=a.data[c];if(d<e)return-1;if(d>e)return 1;if(d!=e){if(!b)return NaN;if(d==d)return 1;if(e==e)return-1}}break;case
7:for(var
c=0;c<this.data.length;c+=2){if(this.data[c+1]<a.data[c+1])return-1;if(this.data[c+1]>a.data[c+1])return 1;if(this.data[c]>>>0<a.data[c]>>>0)return-1;if(this.data[c]>>>0>a.data[c]>>>0)return 1}break;case
2:case
3:case
4:case
5:case
6:case
8:case
9:case
12:for(var
c=0;c<this.data.length;c++){if(this.data[c]<a.data[c])return-1;if(this.data[c]>a.data[c])return 1}break}return 0};function
kv(a,b,c,d){this.kind=a;this.layout=b;this.dims=c;this.data=d}kv.prototype=new
iI();kv.prototype.offset=function(a){if(typeof
a!=="number")if(a
instanceof
Array&&a.length==1)a=a[0];else
aT("Ml_Bigarray_c_1_1.offset");if(a<0||a>=this.dims[0])l1();return a};kv.prototype.get=function(a){return this.data[a]};kv.prototype.set=function(a,b){this.data[a]=b;return 0};kv.prototype.fill=function(a){this.data.fill(a);return 0};function
Eo(a,b,c,d){var
e=Nn(a);if(q7(c)*e!=d.length)aT("length doesn't match dims");if(b==0&&c.length==1&&e==1)return new
kv(a,b,c,d);return new
iI(a,b,c,d)}function
NF(a){return a.slice(1)}function
awY(a,b,c){var
d=NF(c),e=Nl(a,q7(d));return Eo(a,b,d,e)}function
l2(a,b,c){a.set(a.offset(b),c);return 0}function
l5(a,b,c){var
d=String.fromCharCode;if(b==0&&c<=LI&&c==a.length)return d.apply(null,a);var
f=e;for(;0<c;b+=lt,c-=lt)f+=d.apply(null,a.slice(b,b+Math.min(c,lt)));return f}function
q_(a){var
c=new
Uint8Array(a.l),e=a.c,d=e.length,b=0;for(;b<d;b++)c[b]=e.charCodeAt(b);for(d=a.l;b<d;b++)c[b]=0;a.c=c;a.t=4;return c}function
h_(a,b,c,d,e){if(e==0)return 0;if(d==0&&(e>=c.l||c.t==2&&e>=c.c.length)){c.c=a.t==4?l5(a.c,b,e):b==0&&a.c.length==e?a.c:a.c.substr(b,e);c.t=c.c.length==c.l?0:2}else if(c.t==2&&d==c.c.length){c.c+=a.t==4?l5(a.c,b,e):b==0&&a.c.length==e?a.c:a.c.substr(b,e);c.t=c.c.length==c.l?0:2}else{if(c.t!=4)q_(c);var
g=a.c,h=c.c;if(a.t==4)if(d<=b)for(var
f=0;f<e;f++)h[d+f]=g[b+f];else
for(var
f=e-1;f>=0;f--)h[d+f]=g[b+f];else{var
i=Math.min(e,g.length-b);for(var
f=0;f<i;f++)h[d+f]=g.charCodeAt(b+f);for(;f<e;f++)h[d+f]=0}}return 0}function
kA(a,b){if(a==0)return e;if(b.repeat)return b.repeat(a);var
d=e,c=0;for(;;){if(a&1)d+=b;a>>=1;if(a==0)return d;b+=b;c++;if(c==9)b.slice(0,1)}}function
q$(a){if(a.t==2)a.c+=kA(a.l-a.c.length,"\0");else
a.c=l5(a.c,0,a.c.length);a.t=0}function
EM(a){if(a.length<24){for(var
b=0;b<a.length;b++)if(a.charCodeAt(b)>aa)return false;return true}else
return!/[^\x00-\x7f]/.test(a)}function
NS(a){for(var
k=e,d=e,h,g,i,b,c=0,j=a.length;c<j;c++){g=a.charCodeAt(c);if(g<ch){for(var
f=c+1;f<j&&(g=a.charCodeAt(f))<ch;f++);if(f-c>wh){d.substr(0,1);k+=d;d=e;k+=a.slice(c,f)}else
d+=a.slice(c,f);if(f==j)break;c=f}b=1;if(++c<j&&((i=a.charCodeAt(c))&-64)==aA){h=i+(g<<6);if(g<MM){b=h-0x3080;if(b<ch)b=1}else{b=2;if(++c<j&&((i=a.charCodeAt(c))&-64)==aA){h=i+(h<<6);if(g<JN){b=h-0xe2080;if(b<Mq||b>=0xd7ff&&b<0xe000)b=2}else{b=3;if(++c<j&&((i=a.charCodeAt(c))&-64)==aA&&g<0xf5){b=i-0x3c82080+(h<<6);if(b<0x10000||b>0x10ffff)b=3}}}}}if(b<4){c-=b;d+="\ufffd"}else if(b>cz)d+=String.fromCharCode(0xd7c0+(b>>10),KN+(b&0x3FF));else
d+=String.fromCharCode(b);if(d.length>lt){d.substr(0,1);k+=d;d=e}}return k+d}function
h9(a,b,c){this.t=a;this.c=b;this.l=c}h9.prototype.toString=function(){switch(this.t){case
9:return this.c;default:q$(this);case
0:if(EM(this.c)){this.t=9;return this.c}this.t=8;case
8:return this.c}};h9.prototype.toUtf16=function(){var
a=this.toString();if(this.t==9)return a;return NS(a)};h9.prototype.slice=function(){var
a=this.t==4?this.c.slice():this.c;return new
h9(this.t,a,this.l)};function
Nq(a){return new
h9(0,a,a.length)}function
ib(a){return a}function
ci(a){return Nq(ib(a))}function
hE(a,b,c,d,e){h_(ci(a),b,c,d,e);return 0}function
ky(a){return new
N(a[7]<<0|a[6]<<8|a[5]<<16,a[4]<<0|a[3]<<8|a[2]<<16,a[1]<<0|a[0]<<8)}function
bx(a,b){switch(a.t&6){default:if(b>=a.c.length)return 0;case
0:return a.c.charCodeAt(b);case
4:return a.c[b]}}function
Ep(){aT(wa)}function
aw3(a,b){if(b>>>0>=a.l-7)Ep();var
d=new
Array(8);for(var
c=0;c<8;c++)d[7-c]=bx(a,b+c);return ky(d)}function
ab(a,b,c){c&=h4;if(a.t!=4){if(b==a.c.length){a.c+=String.fromCharCode(c);if(b+1==a.l)a.t=0;return 0}q_(a)}a.c[b]=c;return 0}function
h$(a,b,c){if(b>>>0>=a.l)Ep();return ab(a,b,c)}function
kz(a){return a.toArray()}function
aw4(a,b,c){if(b>>>0>=a.l-7)Ep();var
e=kz(c);for(var
d=0;d<8;d++)ab(a,b+7-d,e[d]);return 0}function
hF(d,c){var
f=d.l>=0?d.l:d.l=d.length,e=c.length,b=f-e;if(b==0)return d.apply(null,c);else if(b<0){var
a=d.apply(null,c.slice(0,f));if(typeof
a!=="function")return a;return hF(a,c.slice(f))}else{switch(b){case
1:{var
a=function(a){var
f=new
Array(e+1);for(var
b=0;b<e;b++)f[b]=c[b];f[e]=a;return d.apply(null,f)};break}case
2:{var
a=function(a,b){var
g=new
Array(e+2);for(var
f=0;f<e;f++)g[f]=c[f];g[e]=a;g[e+1]=b;return d.apply(null,g)};break}default:var
a=function(){var
e=arguments.length==0?1:arguments.length,b=new
Array(c.length+e);for(var
a=0;a<c.length;a++)b[a]=c[a];for(var
a=0;a<arguments.length;a++)b[c.length+a]=arguments[a];return hF(d,b)}}a.l=b;return a}}function
at(a,b){if(b>>>0>=a.length-1)l1();return a}function
aw5(a){if(isFinite(a)){if(Math.abs(a)>=2.2250738585072014e-308)return 0;if(a!=0)return 1;return 2}return isNaN(a)?4:3}function
axo(a){return a==CG?1:0}var
axS=Math.log2&&Math.log2(1.1235582092889474E+307)==1020;function
axR(a){if(axS)return Math.floor(Math.log2(a));var
b=0;if(a==0)return-Infinity;if(a>=1)while(a>=2){a/=2;b++}else
while(a<1){a*=2;b--}return b}function
Ev(a){var
b=new
Float32Array(1);b[0]=a;var
c=new
Int32Array(b.buffer);return c[0]|0}function
b3(a,b,c){return new
N(a,b,c)}function
rc(a){if(!isFinite(a)){if(isNaN(a))return b3(1,0,Lr);return a>0?b3(0,0,Lr):b3(0,0,0xfff0)}var
f=a==0&&1/a==-Infinity?cx:a>=0?0:cx;if(f)a=-a;var
b=axR(a)+1023;if(b<=0){b=0;a/=Math.pow(2,-Kl)}else{a/=Math.pow(2,b-Ku);if(a<16){a*=2;b-=1}if(b==0)a/=2}var
d=Math.pow(2,24),c=a|0;a=(a-c)*d;var
e=a|0;a=(a-e)*d;var
g=a|0;c=c&Au|f|b<<4;return b3(g,e,c)}function
Np(a,b,c){a.write(32,b.dims.length);a.write(32,b.kind|b.layout<<8);if(b.caml_custom==pr)for(var
d=0;d<b.dims.length;d++)if(b.dims[d]<cz)a.write(16,b.dims[d]);else{a.write(16,cz);a.write(32,0);a.write(32,b.dims[d])}else
for(var
d=0;d<b.dims.length;d++)a.write(32,b.dims[d]);switch(b.kind){case
2:case
3:case
12:for(var
d=0;d<b.data.length;d++)a.write(8,b.data[d]);break;case
4:case
5:for(var
d=0;d<b.data.length;d++)a.write(16,b.data[d]);break;case
6:for(var
d=0;d<b.data.length;d++)a.write(32,b.data[d]);break;case
8:case
9:a.write(8,0);for(var
d=0;d<b.data.length;d++)a.write(32,b.data[d]);break;case
7:for(var
d=0;d<b.data.length/2;d++){var
f=kz(b.get(d));for(var
e=0;e<8;e++)a.write(8,f[e])}break;case
1:for(var
d=0;d<b.data.length;d++){var
f=kz(rc(b.get(d)));for(var
e=0;e<8;e++)a.write(8,f[e])}break;case
0:for(var
d=0;d<b.data.length;d++){var
f=Ev(b.get(d));a.write(32,f)}break;case
10:for(var
d=0;d<b.data.length/2;d++){var
e=b.get(d);a.write(32,Ev(e[1]));a.write(32,Ev(e[2]))}break;case
11:for(var
d=0;d<b.data.length/2;d++){var
g=b.get(d),f=kz(rc(g[1]));for(var
e=0;e<8;e++)a.write(8,f[e]);var
f=kz(rc(g[2]));for(var
e=0;e<8;e++)a.write(8,f[e])}break}c[0]=(4+b.dims.length)*4;c[1]=(4+b.dims.length)*8}function
Ew(a){var
b=new
Int32Array(1);b[0]=a;var
c=new
Float32Array(b.buffer);return c[0]}function
Ex(a){var
f=a.lo,g=a.mi,c=a.hi,d=(c&0x7fff)>>4;if(d==Jn)return(f|g|c&Au)==0?c&cx?-Infinity:Infinity:NaN;var
e=Math.pow(2,-24),b=(f*e+g)*e+(c&Au);if(d>0){b+=16;b*=Math.pow(2,d-Ku)}else
b*=Math.pow(2,-Kl);if(c&cx)b=-b;return b}function
bp(a){if(!bq.Failure)bq.Failure=[ah,b5(Ej),-3];EI(bq.Failure,a)}function
Nm(a,b,c){var
k=a.read32s();if(k<0||k>16)bp("input_value: wrong number of bigarray dimensions");var
r=a.read32s(),l=r&h4,q=r>>8&1,j=[];if(c==pr)for(var
d=0;d<k;d++){var
p=a.read16u();if(p==cz){var
u=a.read32u(),v=a.read32u();if(u!=0)bp("input_value: bigarray dimension overflow in 32bit");p=v}j.push(p)}else
for(var
d=0;d<k;d++)j.push(a.read32u());var
f=q7(j),h=Nl(l,f),i=Eo(l,q,j,h);switch(l){case
2:for(var
d=0;d<f;d++)h[d]=a.read8s();break;case
3:case
12:for(var
d=0;d<f;d++)h[d]=a.read8u();break;case
4:for(var
d=0;d<f;d++)h[d]=a.read16s();break;case
5:for(var
d=0;d<f;d++)h[d]=a.read16u();break;case
6:for(var
d=0;d<f;d++)h[d]=a.read32s();break;case
8:case
9:var
t=a.read8u();if(t)bp("input_value: cannot read bigarray with 64-bit OCaml ints");for(var
d=0;d<f;d++)h[d]=a.read32s();break;case
7:var
g=new
Array(8);for(var
d=0;d<f;d++){for(var
e=0;e<8;e++)g[e]=a.read8u();var
s=ky(g);i.set(d,s)}break;case
1:var
g=new
Array(8);for(var
d=0;d<f;d++){for(var
e=0;e<8;e++)g[e]=a.read8u();var
m=Ex(ky(g));i.set(d,m)}break;case
0:for(var
d=0;d<f;d++){var
m=Ew(a.read32s());i.set(d,m)}break;case
10:for(var
d=0;d<f;d++){var
o=Ew(a.read32s()),n=Ew(a.read32s());i.set(d,[kr,o,n])}break;case
11:var
g=new
Array(8);for(var
d=0;d<f;d++){for(var
e=0;e<8;e++)g[e]=a.read8u();var
o=Ex(ky(g));for(var
e=0;e<8;e++)g[e]=a.read8u();var
n=Ex(ky(g));i.set(d,[kr,o,n])}break}b[0]=(4+k)*4;return Eo(l,q,j,h)}function
Nk(a,b,c){return a.compare(b,c)}function
NI(a,b){return Math.imul(a,b)}function
ia(a,b){b=NI(b,0xcc9e2d51|0);b=b<<15|b>>>32-15;b=NI(b,0x1b873593);a^=b;a=a<<13|a>>>32-13;return(a+(a<<2)|0)+(0xe6546b64|0)|0}function
axa(a,b){a=ia(a,Ez(b));a=ia(a,Ey(b));return a}function
Nz(a,b){return axa(a,rc(b))}function
No(a){var
c=q7(a.dims),d=0;switch(a.kind){case
2:case
3:case
12:if(c>lD)c=lD;var
e=0,b=0;for(b=0;b+4<=a.data.length;b+=4){e=a.data[b+0]|a.data[b+1]<<8|a.data[b+2]<<16|a.data[b+3]<<24;d=ia(d,e)}e=0;switch(c&3){case
3:e=a.data[b+2]<<16;case
2:e|=a.data[b+1]<<8;case
1:e|=a.data[b+0];d=ia(d,e)}break;case
4:case
5:if(c>aA)c=aA;var
e=0,b=0;for(b=0;b+2<=a.data.length;b+=2){e=a.data[b+0]|a.data[b+1]<<16;d=ia(d,e)}if((c&1)!=0)d=ia(d,a.data[b]);break;case
6:if(c>64)c=64;for(var
b=0;b<c;b++)d=ia(d,a.data[b]);break;case
8:case
9:if(c>64)c=64;for(var
b=0;b<c;b++)d=ia(d,a.data[b]);break;case
7:if(c>32)c=32;c*=2;for(var
b=0;b<c;b++)d=ia(d,a.data[b]);break;case
10:c*=2;case
0:if(c>64)c=64;for(var
b=0;b<c;b++)d=Nz(d,a.data[b]);break;case
11:c*=2;case
1:if(c>32)c=32;for(var
b=0;b<c;b++)d=Nz(d,a.data[b]);break}return d}function
axc(a,b){b[0]=4;return a.read32s()}function
axF(a,b){switch(a.read8u()){case
1:b[0]=4;return a.read32s();case
2:bp("input_value: native integer value too large");default:bp("input_value: ill-formed native integer")}}function
axm(a,b){var
d=new
Array(8);for(var
c=0;c<8;c++)d[c]=a.read8u();b[0]=8;return ky(d)}function
axi(a,b,c){var
e=kz(b);for(var
d=0;d<8;d++)a.write(8,e[d]);c[0]=8;c[1]=8}function
axd(a,b,c){return a.compare(b)}function
axf(a){return a.lo32()^a.hi32()}var
Nu={"_j":{deserialize:axm,serialize:axi,fixed_length:8,compare:axd,hash:axf},"_i":{deserialize:axc,fixed_length:4},"_n":{deserialize:axF,fixed_length:4},"_bigarray":{deserialize:function(a,b){return Nm(a,b,"_bigarray")},serialize:Np,compare:Nk,hash:No},"_bigarr02":{deserialize:function(a,b){return Nm(a,b,pr)},serialize:Np,compare:Nk,hash:No}};function
Er(a){return Nu[a.caml_custom]&&Nu[a.caml_custom].compare}function
Ns(a,b,c,d){var
f=Er(b);if(f){var
e=c>0?f(b,a,d):f(a,b,d);if(d&&e!=e)return c;if(+e!=+e)return+e;if((e|0)!=0)return e|0}return c}function
ED(a){return typeof
a==="string"&&!/[^\x00-\xff]/.test(a)}function
EC(a){return a
instanceof
h9}function
Nt(a){if(typeof
a==="number")return lY;else if(EC(a))return kc;else if(ED(a))return 1252;else if(a
instanceof
Array&&a[0]===a[0]>>>0&&a[0]<=pI){var
b=a[0]|0;return b==kr?0:b}else if(a
instanceof
String)return LL;else if(typeof
a=="string")return LL;else if(a
instanceof
Number)return lY;else if(a&&a.caml_custom)return AS;else if(a&&a.compare)return 1256;else if(typeof
a=="function")return 1247;else if(typeof
a=="symbol")return 1251;return 1001}function
bV(a,b){if(a<b)return-1;if(a==b)return 0;return 1}function
B(a,b){return a<b?-1:a>b?1:0}function
aw2(a,b){a.t&6&&q$(a);b.t&6&&q$(b);return a.c<b.c?-1:a.c>b.c?1:0}function
q9(a,b,c){var
f=[];for(;;){if(!(c&&a===b)){var
e=Nt(a);if(e==km){a=a[1];continue}var
g=Nt(b);if(g==km){b=b[1];continue}if(e!==g){if(e==lY){if(g==AS)return Ns(a,b,-1,c);return-1}if(g==lY){if(e==AS)return Ns(b,a,1,c);return 1}return e<g?-1:1}switch(e){case
247:aT(DY);break;case
248:var
d=bV(a[2],b[2]);if(d!=0)return d|0;break;case
249:aT(DY);break;case
250:aT("equal: got Forward_tag, should not happen");break;case
251:aT("equal: abstract value");break;case
252:if(a!==b){var
d=aw2(a,b);if(d!=0)return d|0}break;case
253:aT("equal: got Double_tag, should not happen");break;case
254:aT("equal: got Double_array_tag, should not happen");break;case
255:aT("equal: got Custom_tag, should not happen");break;case
1247:aT(DY);break;case
1255:var
i=Er(a);if(i!=Er(b))return a.caml_custom<b.caml_custom?-1:1;if(!i)aT("compare: abstract value");var
d=i(a,b,c);if(d!=d)return c?-1:d;if(d!==(d|0))return-1;if(d!=0)return d|0;break;case
1256:var
d=a.compare(b,c);if(d!=d)return c?-1:d;if(d!==(d|0))return-1;if(d!=0)return d|0;break;case
1000:a=+a;b=+b;if(a<b)return-1;if(a>b)return 1;if(a!=b){if(!c)return NaN;if(a==a)return 1;if(b==b)return-1}break;case
1001:if(a<b)return-1;if(a>b)return 1;if(a!=b){if(!c)return NaN;if(a==a)return 1;if(b==b)return-1}break;case
1251:if(a!==b){if(!c)return NaN;return 1}break;case
1252:var
a=ib(a),b=ib(b);if(a!==b){if(a<b)return-1;if(a>b)return 1}break;case
12520:var
a=a.toString(),b=b.toString();if(a!==b){if(a<b)return-1;if(a>b)return 1}break;case
246:case
254:default:if(axo(e)){aT("compare: continuation value");break}if(a.length!=b.length)return a.length<b.length?-1:1;if(a.length>1)f.push(a,b,1);break}}if(f.length==0)return 0;var
h=f.pop();b=f.pop();a=f.pop();if(h+1<a.length)f.push(a,b,h+1);a=a[h];b=b[h]}}function
Nr(a,b){return q9(a,b,true)}function
aw6(){return[0]}function
ar(a){if(a<0)aT("Bytes.create");return new
h9(a?2:9,e,a)}var
ra=[0];function
aw7(a,b){if(ra!==a)return 0;ra=b;return 1}function
Nv(a){return ra}function
aw8(a){ra=a}function
kx(a,b){return+(q9(a,b,false)==0)}function
aw_(a,b,c,d){if(c>0)if(b==0&&(c>=a.l||a.t==2&&c>=a.c.length))if(d==0){a.c=e;a.t=2}else{a.c=kA(c,String.fromCharCode(d));a.t=c==a.l?0:2}else{if(a.t!=4)q_(a);for(c+=b;b<c;b++)a.c[b]=d}return 0}function
Et(a){var
b;a=ib(a);b=+a;if(a.length>0&&b===b)return b;a=a.replace(/_/g,e);b=+a;if(a.length>0&&b===b||/^[+-]?nan$/i.test(a))return b;var
c=/^ *([+-]?)0x([0-9a-f]+)\.?([0-9a-f]*)(p([+-]?[0-9]+))?/i.exec(a);if(c){var
d=c[3].replace(/0+$/,e),g=parseInt(c[1]+c[2]+d,16),f=(c[5]|0)-4*d.length;b=g*Math.pow(2,f);return b}if(/^\+?inf(inity)?$/i.test(a))return Infinity;if(/^-inf(inity)?$/i.test(a))return-Infinity;bp("float_of_string")}function
EH(a){a=ib(a);var
e=a.length;if(e>31)aT("format_int: format too long");var
b={justify:hj,signstyle:bP,filler:hb,alternate:false,base:0,signedconv:false,width:0,uppercase:false,sign:1,prec:-1,conv:LK};for(var
d=0;d<e;d++){var
c=a.charAt(d);switch(c){case"-":b.justify=bP;break;case"+":case" ":b.signstyle=c;break;case"0":b.filler=bm;break;case"#":b.alternate=true;break;case"1":case"2":case"3":case"4":case"5":case"6":case"7":case"8":case"9":b.width=0;while(c=a.charCodeAt(d)-48,c>=0&&c<=9){b.width=b.width*10+c;d++}d--;break;case".":b.prec=0;d++;while(c=a.charCodeAt(d)-48,c>=0&&c<=9){b.prec=b.prec*10+c;d++}d--;case"d":case"i":b.signedconv=true;case"u":b.base=10;break;case"x":b.base=16;break;case"X":b.base=16;b.uppercase=true;break;case"o":b.base=8;break;case"e":case"f":case"g":b.signedconv=true;b.conv=c;break;case"E":case"F":case"G":b.signedconv=true;b.uppercase=true;b.conv=c.toLowerCase();break}}return b}function
Es(a,b){if(a.uppercase)b=b.toUpperCase();var
f=b.length;if(a.signedconv&&(a.sign<0||a.signstyle!=bP))f++;if(a.alternate){if(a.base==8)f+=1;if(a.base==16)f+=2}var
c=e;if(a.justify==hj&&a.filler==hb)for(var
d=f;d<a.width;d++)c+=hb;if(a.signedconv)if(a.sign<0)c+=bP;else if(a.signstyle!=bP)c+=a.signstyle;if(a.alternate&&a.base==8)c+=bm;if(a.alternate&&a.base==16)c+=a.uppercase?"0X":iE;if(a.justify==hj&&a.filler==bm)for(var
d=f;d<a.width;d++)c+=bm;c+=b;if(a.justify==bP)for(var
d=f;d<a.width;d++)c+=hb;return b5(c)}function
Eu(a,b){function
j(a,b){if(Math.abs(a)<1.0)return a.toFixed(b);else{var
c=parseInt(a.toString().split(hj)[1]);if(c>20){c-=20;a/=Math.pow(10,c);a+=new
Array(c+1).join(bm);if(b>0)a=a+cw+new
Array(b+1).join(bm);return a}else
return a.toFixed(b)}}var
c,f=EH(a),e=f.prec<0?6:f.prec;if(b<0||b==0&&1/b==-Infinity){f.sign=-1;b=-b}if(isNaN(b)){c=yI;f.filler=hb}else if(!isFinite(b)){c="inf";f.filler=hb}else
switch(f.conv){case"e":var
c=b.toExponential(e),d=c.length;if(c.charAt(d-3)==xP)c=c.slice(0,d-1)+bm+c.slice(d-1);break;case"f":c=j(b,e);break;case"g":e=e?e:1;c=b.toExponential(e-1);var
i=c.indexOf(xP),h=+c.slice(i+1);if(h<-4||b>=1e21||b.toFixed(0).length>e){var
d=i-1;while(c.charAt(d)==bm)d--;if(c.charAt(d)==cw)d--;c=c.slice(0,d+1)+c.slice(i);d=c.length;if(c.charAt(d-3)==xP)c=c.slice(0,d-1)+bm+c.slice(d-1);break}else{var
g=e;if(h<0){g-=h+1;c=b.toFixed(g)}else
while(c=b.toFixed(g),c.length>e+1)g--;if(g){var
d=c.length-1;while(c.charAt(d)==bm)d--;if(c.charAt(d)==cw)d--;c=c.slice(0,d+1)}}break}return Es(f,c)}function
rb(a,b){if(ib(a)==j7)return b5(e+b);var
c=EH(a);if(b<0)if(c.signedconv){c.sign=-1;b=-b}else
b>>>=0;var
d=b.toString(c.base);if(c.prec>=0){c.filler=hb;var
f=c.prec-d.length;if(f>0)d=kA(f,bm)+d}return Es(c,d)}var
NL=0;function
hG(){return NL++}function
Nx(){return[0]}var
rf=[];function
K(a,b,c){var
e=a[1],i=rf[c];if(i===undefined)for(var
h=rf.length;h<c;h++)rf[h]=0;else if(e[i]===b)return e[i-1];var
d=3,g=e[1]*2+1,f;while(d<g){f=d+g>>1|1;if(b<e[f+1])g=f-2;else
d=f}rf[c]=d+1;return b==e[d+1]?e[d]:0}function
axQ(a){for(var
g=e,c=g,b,i,d=0,h=a.length;d<h;d++){b=a.charCodeAt(d);if(b<ch){for(var
f=d+1;f<h&&(b=a.charCodeAt(f))<ch;f++);if(f-d>wh){c.substr(0,1);g+=c;c=e;g+=a.slice(d,f)}else
c+=a.slice(d,f);if(f==h)break;d=f}if(b<Mq){c+=String.fromCharCode(0xc0|b>>6);c+=String.fromCharCode(ch|b&iD)}else if(b<0xd800||b>=LQ)c+=String.fromCharCode(MM|b>>12,ch|b>>6&iD,ch|b&iD);else if(b>=0xdbff||d+1==h||(i=a.charCodeAt(d+1))<KN||i>LQ)c+="\xef\xbf\xbd";else{d++;b=(b<<10)+i-0x35fdc00;c+=String.fromCharCode(JN|b>>18,ch|b>>12&iD,ch|b>>6&iD,ch|b&iD)}if(c.length>lt){c.substr(0,1);g+=c;c=e}}return g+c}function
cl(a){return EM(a)?b5(a):b5(axQ(a))}function
axb(a,b,c){if(!isFinite(a)){if(isNaN(a))return cl(yI);return cl(a>0?Mt:"-infinity")}var
k=a==0&&1/a==-Infinity?1:a>=0?0:1;if(k)a=-a;var
f=0;if(a==0);else if(a<1)while(a<1&&f>-1022){a*=2;f--}else
while(a>=2){a/=2;f++}var
l=f<0?e:hj,g=e;if(k)g=bP;else
switch(c){case
43:g=hj;break;case
32:g=hb;break;default:break}if(b>=0&&b<13){var
i=Math.pow(2,b*4);a=Math.round(a*i)/i}var
d=a.toString(16);if(b>=0){var
j=d.indexOf(cw);if(j<0)d+=cw+kA(b,bm);else{var
h=j+1+b;if(d.length<h)d+=kA(h-d.length,bm);else
d=d.substr(0,h)}}return cl(g+iE+d+"p"+l+f.toString(10))}function
axh(a){return+a.isZero()}function
l4(a){return new
N(a&gr,a>>24&gr,a>>31&cz)}function
axl(a){return a.toInt()}function
axg(a){return+a.isNeg()}function
EA(a){return a.neg()}function
NB(a,b){var
c=EH(a);if(c.signedconv&&axg(b)){c.sign=-1;b=EA(b)}var
d=e,i=l4(c.base),h="0123456789abcdef";do{var
g=b.udivmod(i);b=g.quotient;d=h.charAt(axl(g.modulus))+d}while(!axh(b));if(c.prec>=0){c.filler=hb;var
f=c.prec-d.length;if(f>0)d=kA(f,bm)+d}return Es(c,d)}function
F(a){return a.length}function
w(a,b){return a.charCodeAt(b)}function
NA(a,b){return a.add(b)}function
NC(a,b){return a.mul(b)}function
EB(a,b){return a.ucompare(b)<0}function
NM(a){var
b=0,e=F(a),c=10,d=1;if(e>0)switch(w(a,b)){case
45:b++;d=-1;break;case
43:b++;d=1;break}if(b+1<e&&w(a,b)==48)switch(w(a,b+1)){case
120:case
88:c=16;b+=2;break;case
111:case
79:c=8;b+=2;break;case
98:case
66:c=2;b+=2;break;case
117:case
85:b+=2;break}return[b,d,c]}function
rg(a){if(a>=48&&a<=57)return a-48;if(a>=65&&a<=90)return a-55;if(a>=97&&a<=ai)return a-87;return-1}function
jB(a){var
g=NM(a),f=g[0],i=g[1],d=g[2],h=l4(d),j=new
N(gr,0xfffffff,cz).udivmod(h).quotient,e=w(a,f),b=rg(e);if(b<0||b>=d)bp(h5);var
c=l4(b);for(;;){f++;e=w(a,f);if(e==95)continue;b=rg(e);if(b<0||b>=d)break;if(EB(j,c))bp(h5);b=l4(b);c=NA(NC(h,c),b);if(EB(c,b))bp(h5)}if(f!=F(a))bp(h5);if(d==10&&EB(new
N(0,0,cx),c))bp(h5);if(i<0)c=EA(c);return c}function
NE(a,b){return a.or(b)}function
rd(a){return a.toFloat()}function
b4(a){var
h=NM(a),d=h[0],i=h[1],e=h[2],g=F(a),j=-1>>>0,f=d<g?w(a,d):0,c=rg(f);if(c<0||c>=e)bp(h5);var
b=c;for(d++;d<g;d++){f=w(a,d);if(f==95)continue;c=rg(f);if(c<0||c>=e)break;b=e*b+c;if(b>j)bp(h5)}if(d!=g)bp(h5);b=i*b;if(e==10&&(b|0)!=b)bp(h5);return b|0}function
L(a){if(EM(a))return a;return NS(a)}function
axr(a){var
c={};for(var
b=1;b<a.length;b++){var
d=a[b];c[L(d[1])]=d[2]}return c}var
q8=hF;function
axq(a){return a.l>=0?a.l:a.l=a.length}function
axs(a){return function(){var
d=axq(a),c=new
Array(d);for(var
b=0;b<d;b++)c[b]=arguments[b];return q8(a,c)}}function
EG(a,b,c){if(a[0]==b){a[0]=c;return 1}return 0}function
axt(a){EG(a,jg,j9);return 0}function
axu(a){return a
instanceof
Array&&a[0]==a[0]>>>0&&EG(a,j9,jg)?0:1}function
axv(a){EG(a,jg,km);return 0}function
axw(a,b){return+(q9(a,b,false)<0)}function
NK(a){return b5(a)}function
aw0(a,b){return a.get(a.offset(b))}function
axn(a,b){return a.xor(b)}function
axk(a,b){return a.shift_right_unsigned(b)}function
axj(a,b){return a.shift_left(b)}function
re(a){function
o(a,b){return axj(a,b)}function
h(a,b){return axk(a,b)}function
s(a,b){return NE(a,b)}function
f(a,b){return axn(a,b)}function
k(a,b){return NA(a,b)}function
i(a,b){return NC(a,b)}function
m(a,b){return s(o(a,b),h(a,64-b))}function
g(a,b){return aw0(a,b)}function
j(a,b,c){return l2(a,b,c)}var
q=jB(NK("0xd1342543de82ef95")),l=jB(NK("0xdaba0b6eb09322e3")),b,d,c,e=a,r=g(e,0),n=g(e,1),p=g(e,2),t=g(e,3);b=k(n,p);b=i(f(b,h(b,32)),l);b=i(f(b,h(b,32)),l);b=f(b,h(b,32));j(e,1,k(i(n,q),r));var
d=p,c=t;c=f(c,d);d=m(d,24);d=f(f(d,c),o(c,16));c=m(c,37);j(e,2,d);j(e,3,c);return b}function
iJ(a,b){if(a<0)l1();var
a=a+1|0,c=new
Array(a);c[0]=0;for(var
d=1;d<a;d++)c[d]=b;return c}function
awS(){var
a=new
ArrayBuffer(64),b=new
Uint32Array(a),c=new
Uint8Array(a);return{len:0,w:new
Uint32Array([0x67452301,0xEFCDAB89,0x98BADCFE,0x10325476]),b32:b,b8:c}}var
q5=function(){function
k(a,b){return a+b|0}function
l(a,b,c,d,e,f){b=k(k(b,a),k(d,f));return k(b<<e|b>>>32-e,c)}function
g(a,b,c,d,e,f,g){return l(b&c|~b&d,a,b,e,f,g)}function
h(a,b,c,d,e,f,g){return l(b&d|c&~d,a,b,e,f,g)}function
i(a,b,c,d,e,f,g){return l(b^c^d,a,b,e,f,g)}function
j(a,b,c,d,e,f,g){return l(c^(b|~d),a,b,e,f,g)}return function(a,b){var
c=a[0],d=a[1],e=a[2],f=a[3];c=g(c,d,e,f,b[0],7,0xD76AA478);f=g(f,c,d,e,b[1],12,0xE8C7B756);e=g(e,f,c,d,b[2],17,0x242070DB);d=g(d,e,f,c,b[3],22,0xC1BDCEEE);c=g(c,d,e,f,b[4],7,0xF57C0FAF);f=g(f,c,d,e,b[5],12,0x4787C62A);e=g(e,f,c,d,b[6],17,0xA8304613);d=g(d,e,f,c,b[7],22,0xFD469501);c=g(c,d,e,f,b[8],7,0x698098D8);f=g(f,c,d,e,b[9],12,0x8B44F7AF);e=g(e,f,c,d,b[10],17,0xFFFF5BB1);d=g(d,e,f,c,b[11],22,0x895CD7BE);c=g(c,d,e,f,b[12],7,0x6B901122);f=g(f,c,d,e,b[13],12,0xFD987193);e=g(e,f,c,d,b[14],17,0xA679438E);d=g(d,e,f,c,b[15],22,0x49B40821);c=h(c,d,e,f,b[1],5,0xF61E2562);f=h(f,c,d,e,b[6],9,0xC040B340);e=h(e,f,c,d,b[11],14,0x265E5A51);d=h(d,e,f,c,b[0],20,0xE9B6C7AA);c=h(c,d,e,f,b[5],5,0xD62F105D);f=h(f,c,d,e,b[10],9,0x02441453);e=h(e,f,c,d,b[15],14,0xD8A1E681);d=h(d,e,f,c,b[4],20,0xE7D3FBC8);c=h(c,d,e,f,b[9],5,0x21E1CDE6);f=h(f,c,d,e,b[14],9,0xC33707D6);e=h(e,f,c,d,b[3],14,0xF4D50D87);d=h(d,e,f,c,b[8],20,0x455A14ED);c=h(c,d,e,f,b[13],5,0xA9E3E905);f=h(f,c,d,e,b[2],9,0xFCEFA3F8);e=h(e,f,c,d,b[7],14,0x676F02D9);d=h(d,e,f,c,b[12],20,0x8D2A4C8A);c=i(c,d,e,f,b[5],4,0xFFFA3942);f=i(f,c,d,e,b[8],11,0x8771F681);e=i(e,f,c,d,b[11],16,0x6D9D6122);d=i(d,e,f,c,b[14],23,0xFDE5380C);c=i(c,d,e,f,b[1],4,0xA4BEEA44);f=i(f,c,d,e,b[4],11,0x4BDECFA9);e=i(e,f,c,d,b[7],16,0xF6BB4B60);d=i(d,e,f,c,b[10],23,0xBEBFBC70);c=i(c,d,e,f,b[13],4,0x289B7EC6);f=i(f,c,d,e,b[0],11,0xEAA127FA);e=i(e,f,c,d,b[3],16,0xD4EF3085);d=i(d,e,f,c,b[6],23,0x04881D05);c=i(c,d,e,f,b[9],4,0xD9D4D039);f=i(f,c,d,e,b[12],11,0xE6DB99E5);e=i(e,f,c,d,b[15],16,0x1FA27CF8);d=i(d,e,f,c,b[2],23,0xC4AC5665);c=j(c,d,e,f,b[0],6,0xF4292244);f=j(f,c,d,e,b[7],10,0x432AFF97);e=j(e,f,c,d,b[14],15,0xAB9423A7);d=j(d,e,f,c,b[5],21,0xFC93A039);c=j(c,d,e,f,b[12],6,0x655B59C3);f=j(f,c,d,e,b[3],10,0x8F0CCC92);e=j(e,f,c,d,b[10],15,0xFFEFF47D);d=j(d,e,f,c,b[1],21,0x85845DD1);c=j(c,d,e,f,b[8],6,0x6FA87E4F);f=j(f,c,d,e,b[15],10,0xFE2CE6E0);e=j(e,f,c,d,b[6],15,0xA3014314);d=j(d,e,f,c,b[13],21,0x4E0811A1);c=j(c,d,e,f,b[4],6,0xF7537E82);f=j(f,c,d,e,b[11],10,0xBD3AF235);e=j(e,f,c,d,b[2],15,0x2AD7D2BB);d=j(d,e,f,c,b[9],21,0xEB86D391);a[0]=k(c,a[0]);a[1]=k(d,a[1]);a[2]=k(e,a[2]);a[3]=k(f,a[3])}}();function
awT(a,b,c){var
e=a.len&iD,d=0;a.len+=c;if(e){var
f=64-e;if(c<f){a.b8.set(b.subarray(0,c),e);return}a.b8.set(b.subarray(0,f),e);q5(a.w,a.b32);c-=f;d+=f}while(c>=64){a.b8.set(b.subarray(d,d+64),0);q5(a.w,a.b32);c-=64;d+=64}if(c)a.b8.set(b.subarray(d,d+c),0)}function
awR(a){var
c=a.len&iD;a.b8[c]=ch;c++;if(c>56){for(var
b=c;b<64;b++)a.b8[b]=0;q5(a.w,a.b32);for(var
b=0;b<56;b++)a.b8[b]=0}else
for(var
b=c;b<56;b++)a.b8[b]=0;a.b32[14]=a.len<<3;a.b32[15]=a.len>>29&0x1FFFFFFF;q5(a.w,a.b32);var
e=new
Uint8Array(16);for(var
d=0;d<4;d++)for(var
b=0;b<4;b++)e[d*4+b]=a.w[d]>>8*b&0xFF;return e}function
EL(a){if(a.t!=4)q_(a);return a.c}function
axL(a){return b5(l5(a,0,a.length))}function
axy(a,b,c){var
d=awS(),e=EL(a);awT(d,e.subarray(b,b+c),c);return axL(awR(d))}function
axz(a,b,c){return axy(ci(a),b,c)}function
cj(a){return a.l}function
axA(){return 0}function
Z(a){EI(bq.Sys_error,a)}var
ic=new
Array();function
cI(a){var
b=ic[a];if(!b.opened)Z("Cannot flush a closed channel");if(!b.buffer||b.buffer_curr==0)return 0;if(b.output)b.output(l5(b.buffer,0,b.buffer_curr));else
b.file.write(b.offset,b.buffer,0,b.buffer_curr);b.offset+=b.buffer_curr;b.buffer_curr=0;return 0}function
Nj(){}function
axY(a){var
d=F(a),c=new
Uint8Array(d),b=0;for(;b<d;b++)c[b]=w(a,b);return c}function
cH(a,b){this.fs=require(yT);this.fd=a;this.flags=b}cH.prototype=new
Nj();cH.prototype.constructor=cH;cH.prototype.truncate=function(a){try{this.fs.ftruncateSync(this.fd,a|0)}catch(f){Z(f.toString())}};cH.prototype.length=function(){try{return this.fs.fstatSync(this.fd).size}catch(f){Z(f.toString())}};cH.prototype.write=function(a,b,c,d){try{if(this.flags.isCharacterDevice)this.fs.writeSync(this.fd,b,c,d);else
this.fs.writeSync(this.fd,b,c,d,a)}catch(f){Z(f.toString())}return 0};cH.prototype.read=function(a,b,c,d){try{if(this.flags.isCharacterDevice)var
e=this.fs.readSync(this.fd,b,c,d);else
var
e=this.fs.readSync(this.fd,b,c,d,a);return e}catch(f){Z(f.toString())}};cH.prototype.close=function(){try{this.fs.closeSync(this.fd);return 0}catch(f){Z(f.toString())}};function
axO(a,b){if(b.name)try{var
d=require(yT),c=d.openSync(b.name,"rs");return new
cH(c,b)}catch(f){}return new
cH(a,b)}var
ri=new
Array(3);function
l6(){return typeof
af.process!=="undefined"&&typeof
af.process.versions!=="undefined"&&typeof
af.process.versions.node!=="undefined"}function
axT(){function
a(a){if(a.charAt(0)===bv)return[e,a.substring(1)];return}function
b(a){var
h=/^([a-zA-Z]:|[\\/]{2}[^\\/]+[\\/]+[^\\/]+)?([\\/])?([\s\S]*?)$/,b=h.exec(a),c=b[1]||e,f=Boolean(c&&c.charAt(1)!==jy);if(Boolean(b[2]||f)){var
d=b[1]||e,g=b[2]||e;return[d,a.substring(d.length+g.length)]}return}return l6()&&af.process&&af.process.platform?af.process.platform===Ke?b:a:a}var
EN=axT();function
NR(a){return a.slice(-1)!==bv?a+bv:a}if(l6()&&af.process&&af.process.cwd)var
l3=af.process.cwd().replace(/\\/g,bv);else
var
l3="/static";l3=NR(l3);function
axx(a){a=L(a);if(!EN(a))a=l3+a;var
e=EN(a),d=e[1].split(bv),b=[];for(var
c=0;c<d.length;c++)switch(d[c]){case"..":if(b.length>1)b.pop();break;case".":break;case"":break;default:b.push(d[c]);break}b.unshift(e[0]);b.orig=a;return b}var
axW=["E2BIG","EACCES","EAGAIN",tl,"EBUSY","ECHILD","EDEADLK","EDOM",Ne,"EFAULT","EFBIG","EINTR","EINVAL","EIO","EISDIR","EMFILE","EMLINK","ENAMETOOLONG","ENFILE","ENODEV",At,"ENOEXEC","ENOLCK","ENOMEM","ENOSPC","ENOSYS",vd,J0,"ENOTTY","ENXIO","EPERM","EPIPE","ERANGE","EROFS","ESPIPE","ESRCH","EXDEV","EWOULDBLOCK","EINPROGRESS","EALREADY","ENOTSOCK","EDESTADDRREQ","EMSGSIZE","EPROTOTYPE","ENOPROTOOPT","EPROTONOSUPPORT","ESOCKTNOSUPPORT","EOPNOTSUPP","EPFNOSUPPORT","EAFNOSUPPORT","EADDRINUSE","EADDRNOTAVAIL","ENETDOWN","ENETUNREACH","ENETRESET","ECONNABORTED","ECONNRESET","ENOBUFS","EISCONN","ENOTCONN","ESHUTDOWN","ETOOMANYREFS","ETIMEDOUT","ECONNREFUSED","EHOSTDOWN","EHOSTUNREACH","ELOOP","EOVERFLOW"];function
ie(a,b,c,d){var
f=axW.indexOf(a);if(f<0){if(d==null)d=-9999;f=[0,d]}var
g=[f,cl(b||e),cl(c||e)];return g}var
NJ={};function
iK(a){return NJ[a]}function
id(a,b){throw x([0,a].concat(b))}function
Eq(a){if(!(a
instanceof
Uint8Array))a=new
Uint8Array(a);return new
h9(4,a,a.length)}function
NO(a){Z(a+o$)}function
bw(a){this.data=a}bw.prototype=new
Nj();bw.prototype.constructor=bw;bw.prototype.truncate=function(a){var
b=this.data;this.data=ar(a|0);h_(b,0,this.data,0,a)};bw.prototype.length=function(){return cj(this.data)};bw.prototype.write=function(a,b,c,d){var
e=this.length();if(a+d>=e){var
f=ar(a+d),g=this.data;this.data=f;h_(g,0,this.data,0,e)}h_(Eq(b),c,this.data,a,d);return 0};bw.prototype.read=function(a,b,c,d){var
e=this.length();if(a+d>=e)d=e-a;if(d){var
f=ar(d|0);h_(this.data,a,f,0,d);b.set(EL(f),c)}return d};function
jA(a,b,c){this.file=b;this.name=a;this.flags=c}jA.prototype.err_closed=function(){Z(this.name+MS)};jA.prototype.length=function(){if(this.file)return this.file.length();this.err_closed()};jA.prototype.write=function(a,b,c,d){if(this.file)return this.file.write(a,b,c,d);this.err_closed()};jA.prototype.read=function(a,b,c,d){if(this.file)return this.file.read(a,b,c,d);this.err_closed()};jA.prototype.close=function(){this.file=undefined};function
a$(a,b){this.content={};this.root=a;this.lookupFun=b}a$.prototype.nm=function(a){return this.root+a};a$.prototype.create_dir_if_needed=function(a){var
d=a.split(bv),c=e;for(var
b=0;b<d.length-1;b++){c+=d[b]+bv;if(this.content[c])continue;this.content[c]=Symbol("directory")}};a$.prototype.slash=function(a){return/\/$/.test(a)?a:a+bv};a$.prototype.lookup=function(a){if(!this.content[a]&&this.lookupFun){var
b=this.lookupFun(b5(this.root),b5(a));if(b!==0){this.create_dir_if_needed(a);this.content[a]=new
bw(ci(b[1]))}}};a$.prototype.exists=function(a){if(a==e)return 1;var
b=this.slash(a);if(this.content[b])return 1;this.lookup(a);return this.content[a]?1:0};a$.prototype.isFile=function(a){return this.exists(a)&&!this.is_dir(a)?1:0};a$.prototype.mkdir=function(a,b,c){var
f=c&&iK(p4);if(this.exists(a))if(f)id(f,ie(Ne,CE,this.nm(a)));else
Z(a+": File exists");var
d=/^(.*)\/[^/]+/.exec(a);d=d&&d[1]||e;if(!this.exists(d))if(f)id(f,ie(At,CE,this.nm(d)));else
Z(d+o$);if(!this.is_dir(d))if(f)id(f,ie(vd,CE,this.nm(d)));else
Z(d+C7);this.create_dir_if_needed(this.slash(a))};a$.prototype.rmdir=function(a,b){var
c=b&&iK(p4),d=a==e?e:this.slash(a),g=new
RegExp(pS+d+M5);if(!this.exists(a))if(c)id(c,ie(At,wg,this.nm(a)));else
Z(a+o$);if(!this.is_dir(a))if(c)id(c,ie(vd,wg,this.nm(a)));else
Z(a+C7);for(var
f
in
this.content)if(f.match(g))if(c)id(c,ie(J0,wg,this.nm(a)));else
Z(this.nm(a)+": Directory not empty");delete
this.content[d]};a$.prototype.readdir=function(a){var
g=a==e?e:this.slash(a);if(!this.exists(a))Z(a+o$);if(!this.is_dir(a))Z(a+C7);var
h=new
RegExp(pS+g+M5),d={},c=[];for(var
f
in
this.content){var
b=f.match(h);if(b&&!d[b[1]]){d[b[1]]=true;c.push(b[1])}}return c};a$.prototype.opendir=function(a,b){var
c=b&&iK(p4),d=this.readdir(a),e=false,f=0;return{readSync:function(){if(e)if(c)id(c,ie(tl,KH,this.nm(a)));else
Z(a+JJ);if(f==d.length)return null;var
b=d[f];f++;return{name:b}},closeSync:function(){if(e)if(c)id(c,ie(tl,KH,this.nm(a)));else
Z(a+JJ);e=true;d=[]}}};a$.prototype.is_dir=function(a){if(a==e)return true;var
b=this.slash(a);return this.content[b]?1:0};a$.prototype.unlink=function(a){var
b=this.content[a]?true:false;delete
this.content[a];return b};a$.prototype.open=function(a,b){var
c;if(b.rdonly&&b.wronly)Z(this.nm(a)+vG);if(b.text&&b.binary)Z(this.nm(a)+C8);this.lookup(a);if(this.content[a]){if(this.is_dir(a))Z(this.nm(a)+Lj);if(b.create&&b.excl)Z(this.nm(a)+wA);c=this.content[a];if(b.truncate)c.truncate()}else if(b.create){this.create_dir_if_needed(a);this.content[a]=new
bw(ar(0));c=this.content[a]}else
NO(this.nm(a));return new
jA(this.nm(a),c,b)};a$.prototype.open=function(a,b){var
c;if(b.rdonly&&b.wronly)Z(this.nm(a)+vG);if(b.text&&b.binary)Z(this.nm(a)+C8);this.lookup(a);if(this.content[a]){if(this.is_dir(a))Z(this.nm(a)+Lj);if(b.create&&b.excl)Z(this.nm(a)+wA);c=this.content[a];if(b.truncate)c.truncate()}else if(b.create){this.create_dir_if_needed(a);this.content[a]=new
bw(ar(0));c=this.content[a]}else
NO(this.nm(a));return new
jA(this.nm(a),c,b)};a$.prototype.register=function(a,b){var
c;if(this.content[a])Z(this.nm(a)+wA);if(EC(b))c=new
bw(b);if(ED(b))c=new
bw(ci(b));else if(b
instanceof
Array)c=new
bw(Eq(b));else if(typeof
b==="string")c=new
bw(Nq(b));else if(b.toString){var
d=ci(cl(b.toString()));c=new
bw(d)}if(c){this.create_dir_if_needed(a);this.content[a]=c}else
Z(this.nm(a)+" : registering file with invalid content type")};a$.prototype.constructor=a$;function
aS(a){this.fs=require(yT);this.root=a}aS.prototype.nm=function(a){return this.root+a};aS.prototype.exists=function(a){try{return this.fs.existsSync(this.nm(a))?1:0}catch(f){return 0}};aS.prototype.isFile=function(a){try{return this.fs.statSync(this.nm(a)).isFile()?1:0}catch(f){Z(f.toString())}};aS.prototype.mkdir=function(a,b,c){try{this.fs.mkdirSync(this.nm(a),{mode:b});return 0}catch(f){this.raise_nodejs_error(f,c)}};aS.prototype.rmdir=function(a,b){try{this.fs.rmdirSync(this.nm(a));return 0}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.readdir=function(a,b){try{return this.fs.readdirSync(this.nm(a))}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.is_dir=function(a){try{return this.fs.statSync(this.nm(a)).isDirectory()?1:0}catch(f){Z(f.toString())}};aS.prototype.unlink=function(a,b){try{var
c=this.fs.existsSync(this.nm(a))?1:0;this.fs.unlinkSync(this.nm(a));return c}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.open=function(a,b,c){var
d=require("constants"),e=0;for(var
h
in
b)switch(h){case"rdonly":e|=d.O_RDONLY;break;case"wronly":e|=d.O_WRONLY;break;case"append":e|=d.O_WRONLY|d.O_APPEND;break;case"create":e|=d.O_CREAT;break;case"truncate":e|=d.O_TRUNC;break;case"excl":e|=d.O_EXCL;break;case"binary":e|=d.O_BINARY;break;case"text":e|=d.O_TEXT;break;case"nonblock":e|=d.O_NONBLOCK;break}try{var
f=this.fs.openSync(this.nm(a),e),g=this.fs.lstatSync(this.nm(a)).isCharacterDevice();b.isCharacterDevice=g;return new
cH(f,b)}catch(f){this.raise_nodejs_error(f,c)}};aS.prototype.rename=function(a,b,c){try{this.fs.renameSync(this.nm(a),this.nm(b))}catch(f){this.raise_nodejs_error(f,c)}};aS.prototype.stat=function(a,b){try{var
c=this.fs.statSync(this.nm(a));return this.stats_from_js(c)}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.lstat=function(a,b){try{var
c=this.fs.lstatSync(this.nm(a));return this.stats_from_js(c)}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.symlink=function(a,b,c,d){try{this.fs.symlinkSync(this.nm(b),this.nm(c),a?"dir":"file");return 0}catch(f){this.raise_nodejs_error(f,d)}};aS.prototype.readlink=function(a,b){try{var
c=this.fs.readlinkSync(this.nm(a),"utf8");return cl(c)}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.opendir=function(a,b){try{return this.fs.opendirSync(this.nm(a))}catch(f){this.raise_nodejs_error(f,b)}};aS.prototype.raise_nodejs_error=function(a,b){var
c=iK(p4);if(b&&c){var
d=ie(a.code,a.syscall,a.path,a.errno);id(c,d)}else
Z(a.toString())};aS.prototype.stats_from_js=function(a){var
b;if(a.isFile())b=0;else if(a.isDirectory())b=1;else if(a.isCharacterDevice())b=2;else if(a.isBlockDevice())b=3;else if(a.isSymbolicLink())b=4;else if(a.isFIFO())b=5;else if(a.isSocket())b=6;return[0,a.dev,a.ino,b,a.mode,a.nlink,a.uid,a.gid,a.rdev,a.size,a.atimeMs,a.mtimeMs,a.ctimeMs]};aS.prototype.constructor=aS;function
Ny(a){var
b=EN(a);if(!b)return;return b[0]+bv}var
rh=Ny(l3)||bp("unable to compute caml_root"),kB=[];if(l6())kB.push({path:rh,device:new
aS(rh)});else
kB.push({path:rh,device:new
a$(rh)});kB.push({path:J6,device:new
a$(J6)});function
axV(a){var
g=axx(a),a=g.
